### PR TITLE
add dark mode support, warnings for big dirs

### PR DESCRIPTION
**Pull request by molsonkiko**
1. All forms will now be colored to match the Notepad++ theme. This is true in dark mode and also with non-dark-mode themes that the user might use (e.g., Vim Dark Blue, Blackboard). However, the background of the DataGridView containing the filename rows is bugged in a very weird way. Hopefully someone can figure that out.
image
2. Added a system for checking if the current directory is very large and could potentially cause tremendously long latency when the user has chosen recursive subdirectory search. Now a warning will pop up if there are more than 5000 files in the current directory tree, so that the user can choose to search only the top directory, or not to search that dir at all.
image

### CODE REVIEW

**Comment by young-developer** [comment]
@molsonkiko I get error after typing: 
image

Stack trace:

See the end of this message for details on invoking 

> just-in-time (JIT) debugging instead of this dialog box.
> 
> ************** Exception Text **************
> System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
>    at System.ThrowHelper.ThrowKeyNotFoundException()
>    at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
>    at NavigateTo.Plugin.Namespace.FrmNavigateTo.FilterCurrentDirectory(Func`2 filterFunc)
>    at NavigateTo.Plugin.Namespace.FrmNavigateTo.FilterDataGrid(String filter)
>    at NavigateTo.Plugin.Namespace.FrmNavigateTo.SearchComboBoxTextChanged(Object sender, EventArgs e)
>    at System.Windows.Forms.Control.OnTextChanged(EventArgs e)
>    at System.Windows.Forms.ComboBox.OnTextChanged(EventArgs e)
>    at System.Windows.Forms.ComboBox.WmReflectCommand(Message& m)
>    at System.Windows.Forms.ComboBox.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
> 
> 
> ************** Loaded Assemblies **************
> mscorlib
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9139.0 built by: NET481REL1LAST_B
>     CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
> ----------------------------------------
> NavigateTo
>     Assembly Version: 2.5.2.0
>     Win32 Version: 2.5.2.0
>     CodeBase: file:///C:/Program%20Files/Notepad++/plugins/NavigateTo/NavigateTo.dll
> ----------------------------------------
> System.Drawing
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9032.0 built by: NET481REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
> ----------------------------------------
> System
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9139.0 built by: NET481REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
> ----------------------------------------
> System.Windows.Forms
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9075.0 built by: NET481REL1LAST_C
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
> ----------------------------------------
> System.Configuration
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9032.0 built by: NET481REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
> ----------------------------------------
> System.Core
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9139.0 built by: NET481REL1LAST_B
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
> ----------------------------------------
> System.Xml
>     Assembly Version: 4.0.0.0
>     Win32 Version: 4.8.9032.0 built by: NET481REL1
>     CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
> ----------------------------------------
> 
> ************** JIT Debugging **************
> To enable just-in-time (JIT) debugging, the .config file for this
> application or computer (machine.config) must have the
> jitDebugging value set in the system.windows.forms section.
> The application must also be compiled with debugging
> enabled.
> 
> For example:
> 
> <configuration>
>     <system.windows.forms jitDebugging="true" />
> </configuration>
> 
> When JIT debugging is enabled, any unhandled exception
> will be sent to the JIT debugger registered on the computer
> rather than be handled by this dialog box.
>

**Comment by molsonkiko** [comment]
Oh no! That's really weird. I've tried pretty much everything I can think of to replicate that bug on my machine and I can't.
Things I've tried include:
1. Just opening a file with subdirectory search turned off
2. Just opening a file with top directory search on but subdir search off
3. Opening a file with subdir search on, but the dir tree is big so I click yes to limit to top dir
4. Opening a file with subdir search on, but the dir tree is big, but I click no and search subdirs anyway
5. Opening a file with subdir search on, but the top dir is big, so I click yes to eliminate both subdir and top dir search
6. Opening a file with top dir search on, but top dir is big, so I click yes to eliminate top dir search
7. Doing most of the above, and then opening the settings form and changing various settings
8. Opening the settings form to turn on fuzzy search before ever opening the navigateto form and then typing in some stuff
9. Turning fuzzy search on or off while the navigateto form is already open
10. Doing various things with a "new 1" file open with subdir search active

**Comment by young-developer** [comment]
@molsonkiko It crashes only if search in top directory is enabled

File: NppNavigateTo/Forms/FrmNavigateTo.cs [hunk 1]
```diff
@@ -72,36 +92,97 @@ private static bool MatchesAllWordsInFilter(string s, string[] words)
...
+            bool userWantsSearchSubdirs = FrmSettings.Settings.GetBoolSetting(Settings.searchInSubDirs);
+            long nextTimeToRefresh = lastDirectoryReloadTimeTicks +
+                10_000_000 * FrmSettings.Settings.GetIntSetting(Settings.secondsBetweenDirectoryScans); // 10 million ticks/sec
+            DirectorySearchLevel searchLevel = subDirSearchAllowances.TryGetValue(currentDirectory, out var oldAllowSearch)
+                ? oldAllowSearch
+                : userWantsSearchSubdirs
+                    ? DirectorySearchLevel.RecurseSubdirs
+                    : DirectorySearchLevel.None;
+            filesInCurrentDirectory = SearchCurrentDirectory(searchLevel, nextTimeToRefresh, currentDirectory);
+            searchLevel = subDirSearchAllowances[currentDirectory];
```
Comment (young-developer) [inline NppNavigateTo/Forms/FrmNavigateTo.cs:175]: @molsonkiko It crashes here with top dir search enabled and sub dir disabled.

Comment (young-developer) [reply NppNavigateTo/Forms/FrmNavigateTo.cs:175]: @molsonkiko It looks like some caching of refresh and dictionary key mismatch.

**Comment by molsonkiko** [comment]
Good catch! Yeah, I'm working on it. Should have fix ready by 7pm (US- Los
Angeles time)

**Pull request merged**